pkg/virt-handler/device-manager/device_controller.go

```diff
@@ -195,12 +202,27 @@ func removeSelectorSpaces(selectorName string) string {
 
 }
 
-func (c *DeviceController) refreshMediatedDevicesTypes() {
-	nodeDesiredMdevTypesList := c.virtConfig.GetDesiredMDEVTypes(c.host)
-	err := c.mdevTypesManager.updateMDEVTypesConfiguration(nodeDesiredMdevTypesList)
+func (c *DeviceController) RefreshMediatedDevicesTypes() {
+	go func() {
+		if c.refreshMediatedDevicesTypes() {
+			c.refreshPermittedDevices()
+		}
+	}()
+}
+
+func (c *DeviceController) refreshMediatedDevicesTypes() bool {
+	requiresDevicePluginsUpdate := false
+	node, err := c.clientset.Nodes().Get(context.Background(), c.host, metav1.GetOptions{})
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to configure the desired mdev types, failed to get node details")
+		return requiresDevicePluginsUpdate
+	}
+	nodeDesiredMdevTypesList := c.virtConfig.GetDesiredMDEVTypes(node)
+	requiresDevicePluginsUpdate, err = c.mdevTypesManager.updateMDEVTypesConfiguration(nodeDesiredMdevTypesList)
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to configure the desired mdev types: %s", strings.Join(nodeDesiredMdevTypesList, ", "))
 	}
+	return requiresDevicePluginsUpdate
```

Comment:
Yea, thanks. I can explicitly return `false` when an error is encountered if this is confusing.

In some cases, we won't be able to remove all mdev types. When configuration changes but there is a running VMI that uses a device. We can only log here and return false, to avoid refreshing the device plugins.

Reply:
I'm fine either way now. I looked closer and requiresDevicePluginsUpdate will for sure always return false if an error occurrs. I mistakenly thought it was possible for an error to get returned and for this boolean to be set to true. that's not the case though.